### DISCUSSION TITLE
ESP32 + mpu9250 (gy-91)

### DISCUSSION THREAD

**Discussion by Gooz74** [micropython/micropython#11327] [ESP32]
Hi all, 

I'm very new to programming with micropython and I'm trying to create a small project where I want to readout the values of a mpu9250 (gy-91) sensor with the ESP32 module.
I've uploaded micropython to the ESP32 and it's running micropython v1.19.1.

I've downloaded the libraries mpu9250.py, mpu6500.py and ak8963.py from the github page of Kevin McAleer and I was trying his code (he works with a pi pico) on the ESP32

This is the code that I have until now, I know it's not much but I still hope to find some help here so I can continue.
The line print(i2c.scan()) returns the values [104, 118]. Therefor I know the sensor is connected correctly

      from_ machine import SoftI2C, Pin
      from math import sqrt, atan2, pi, copysign, sin, cos
      from mpu9250 import MPU9250
      from time import sleep
      
      id=0
      scl = Pin(22)
      sda = Pin(21)
      
      i2c = SoftI2C(scl=scl, sda=sda)
      print(i2c.scan())
      
      m = MPU9250(i2c)

I've gotten to the point where I can see that the sensor is connected correctly to the ESP32 but after that I got the following errors.

    Traceback (most recent call last):
      File "<stdin>", line 13, in <module>
      File "mpu9250.py", line 56, in __init__
      File "ak8963.py", line 76, in __init__
      File "ak8963.py", line 139, in whoami
      File "ak8963.py", line 198, in _register_char
     OSError: [Errno 19] ENODEV

**Comment by bixb922** [comment]
You've probably checked this, but are there pull up resistors on SDA and SCL?

**Comment by Gooz74** [comment]
Hi bixb922,

Thank you for the quick reply. Yes, I've tried that, i've tried following things:
- 3.3v to vin (with and without external pull-up resistors)
- 3.3v to 3.3v (with and without external pull-up resistors)
- 5v to vin (with and without external pull-up resistors)
All with the same result. 

From the schematics of the sensor, I've found that the gy-91 board has also internal pull-up registers as well on the vin as on the 3.3v line.

From piece of code "print(i2c.scan())", I get the result "[104, 118]". therefor It seems the sensors are being seen.
It's when I try to initialize the MPU9250 via "m = MPU9250(i2c)", I get the error.
The initialisation I've tried with and without a specific frequency of (freq = 100000)
